hoomd/write/hdf5.py

```diff
@@ -0,0 +1,225 @@
+        custom.Action.Flags.PRESSURE_TENSOR,
+        custom.Action.Flags.EXTERNAL_FIELD_VIRIAL,
+    )
+
+    _reject_categories = logging.LoggerCategories.any((
+        logging.LoggerCategories.object,
+        logging.LoggerCategories.strings,
+        logging.LoggerCategories.string,
+    ))
+
+    accepted_categories = logging.LoggerCategories.ALL - _reject_categories
+
+    def __init__(self, filename, logger, mode="a"):
+        param_dict = ParameterDict(filename=typeconverter.OnlyTypes(
+            (str, PurePath)),
+                                   logger=logging.Logger,
+                                   mode=str)
+        if (rejects := self._reject_categories
+                & logger.categories) != logging.LoggerCategories["NONE"]:
+            reject_str = logging.LoggerCategories._get_string_list(rejects)
+            raise ValueError(f"Cannot have {reject_str} in logger categories.")
+        param_dict.update({
+            "filename": filename,
+            "logger": logger,
+            "mode": mode
+        })
+        self._param_dict = param_dict
+        self._fh = h5py.File(self.filename, mode=self.mode)
+        self._validate_scheme()
+        self._frame = self._find_frame()
+        self._attached_ = False
+
+    def __del__(self):
+        """Closes file upon destruction."""
+        if hasattr(self, "_fh"):
+            self._fh.close()
+
+    def _setattr_param(self, attr, value):
+        """Makes self._param_dict attributes read only."""
+        raise ValueError("Attribute {} is read-only.".format(attr))
+
+    def attach(self, simulation):
+        self._attached_ = True
+
+    def _attached(self):
+        return self._attached_
+
+    def detach(self):
+        self._attached_ = False
+
+    def act(self, timestep):
+        """Write a new frame of logger data to the HDF5 file."""
+        log_dict = util._dict_flatten(self.logger.log())
+        if self._frame == 0:
+            self._initialize_datasets(log_dict)
+        for key, (value, category) in log_dict.items():
+            if logging.LoggerCategories[category] in self._reject_categories:
+                continue
+            if value is None:
+                continue
+            str_key = "/".join(("hoomd-data",) + key)
+            if str_key not in self._fh:
+                raise RuntimeError(
+                    "The logged quantities cannot change within a file.")
+            dataset = self._fh[str_key]
+            dataset.resize(self._frame + 1, axis=0)
+            dataset[self._frame, ...] = value
+        self._frame += 1
+        self._fh["hoomd-data"].attrs["frames"] = self._frame
+
+    def flush(self):
+        """Write out all data currently buffered in memory.
+
+        Without calling this, data may be stored in the h5py.File object without
+        being written to the on disk file yet.
+        """
+        self._fh.flush()
+
+    def _create_database(self, key: str, shape, dtype, chunk_size):
```

Comment:
```suggestion
    def _create_dataset(self, key: str, shape, dtype, chunk_size):
```
Does this seem like a more appropriate name, given the name of the h5 call in its implementation and context it gets used in?

Reply:
Good point.